File: Client/cshaprClient/cshaprClient/MonteCarloPi.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace cshaprClient
{
    class MonteCarloPi
    {
        public int NumAttemps{get;set;}
        public int hits      {get;set;}
        public void runMC()
        {
            for( int i=0; i<NumAttemps; ++i)
            {
                double x = 0;
                double y = 0;
                //x = Math.
            
            
            
            
            
            }
        }
        public double GetRandomNumber(double minimum, double maximum)
        {
            Random random = new Random();
            return random.NextDouble() * (maximum - minimum) + minimum;
        }
    }
}

File: AndroidClient/app/src/main/java/com/example/myfirstapp/MonteCarloPi.java
package com.example.myfirstapp;

public class MonteCarloPi {
    public static int DoMCPi( int numtries)
    {
        int retval =0;
        int hits =0;
        double  x;
        double  y;
        for( int i =0;i<numtries;++i)
        {
            x = Math.random();
            y = Math.random();
            if(hit(x,y))
            {
                ++hits;
            }
        }
        return retval;
    }
    public static boolean hit(double x, double y)
    {
        boolean retval =false;
        retval = ((x * x) + (y * y)) <= 1;
        return retval;
    }
}
